cmd/containerboot/kube.go

```diff
@@ -69,6 +69,23 @@ func findKeyInKubeSecret(ctx context.Context, secretName string) (string, error)
 // the kube secret secretName.
 func storeDeviceID(ctx context.Context, secretName, deviceID string) error {
 	kubeOnce.Do(initKube)
+
+	// First check if the secret exists at all. Even if running on
+	// kubernetes, we do not necessarily store state in a k8s secret.
+	req, err := http.NewRequest("GET", fmt.Sprintf("/api/v1/namespaces/%s/secrets/%s", kubeNamespace, secretName), nil)
+	if err != nil {
+		return err
+	}
+	resp, err := doKubeRequest(ctx, req)
+	if err != nil {
+		if resp != nil && resp.StatusCode/100 == 4 {
```

Comment:
This `resp.StatusCode/100 == 4` is ... unusual, and I did a bit of a double-take; maybe something like `resp.StatusCode >= 400 && resp.StatusCode <= 499`?

Reply:
Yeah, I felt clever at the time, but in retrospect it's not worth the keystroke savings.